migrations/1619968992683-fiat-balance-notification-to-percent.js

```diff
@@ -0,0 +1,52 @@
+const _ = require('lodash/fp')
+const { saveConfig, loadLatest } = require('../lib/new-settings-loader')
+
+exports.up = function (next) {
+  loadLatest()
+    .then(({ config }) => {
+      const fiatBalance1 = config.notifications_fiatBalanceCassette1
+      const fiatBalance2 = config.notifications_fiatBalanceCassette2
+
+      if (fiatBalance1) {
+        config.notifications_fiatBalanceCassette1 = (100 * (fiatBalance1 / 500)).toFixed(0)
+      }
+      if (fiatBalance2) {
+        config.notifications_fiatBalanceCassette2 = (100 * (fiatBalance2 / 500)).toFixed(0)
+      }
+
+      const {
+        notifications_fiatBalanceCassette1: notifications_fillingPercentageCassette1,
+        notifications_fiatBalanceCassette2: notifications_fillingPercentageCassette2,
+        ...rest
+      } = config
+
+      config = { notifications_fillingPercentageCassette1, notifications_fillingPercentageCassette2, ...rest }
```

Comment:
Just a headsup, you don't need to send the whole config object to saveConfig(), internally the config is merged by that function before getting saved to the DB.

This has some implications:
- if the original config object is { a:1, b:2, c:3 } and you do saveConfig({ a:11, d:4 }) the resulting config is { a:11, b:2, c:3, d:4 }
- there isn't a currently supported way of deleting keys from the config object. Best we can do right now is setting the properties to null

My advice is creating a different variable with all the stuff you want to add or change and then call saveConfig with it instead of copying the whole config object onto itself.

Also, set the old fields to null. It would be something like this:

```
const newConfig = {
  notifications_fillingPercentageCassette1: config.notifications_fiatBalanceCassette1, // or use destructuring
  notifications_fillingPercentageCassette2: config.notifications_fiatBalanceCassette2,
  notifications_fiatBalanceCassette1: null,
  notifications_fiatBalanceCassette2: null,
  // etc
}
```

Reply:
Good point!